src/main/java/address/controller/PersonOverviewController.java

```diff
@@ -117,7 +119,10 @@ private void initialize() {
      */
     @FXML
     private void handleDeletePersons() {
-        final List<ReadOnlyViewablePerson> selected = personListView.getSelectionModel().getSelectedItems();
+        final List<ReadOnlyViewablePerson> selected = new ArrayList<>(personListView.getSelectionModel()
+                                                                                    .getSelectedItems());
+        selected.removeIf(p -> !(p instanceof  ReadOnlyViewablePerson));
```

Comment:
You can try this on master :

1) create a new person
2) select the person and uno
3) now nothing is selected. Press D
4) null is deleted (because selectionItems has a null item)
![image](https://cloud.githubusercontent.com/assets/9070608/16442557/f8e219a8-3e05-11e6-877c-3215f7b9f40f.png)